r2/r2/models/subreddit.py

```diff
@@ -540,6 +540,33 @@ def remove_moderator(self, user, **kwargs):
         hook = hooks.get_hook("subreddit.remove_moderator")
         hook.call(subreddit=self, user=user)
 
+        if user == self.top_moderator_account and len(self.moderators) > 1:
+            from r2.models import ModAction
+            fullperms = self.moderator_permission_class.loads('+all')
+            # cycle through each available moderator
+            for user_id in self.moderators[1:]:
```

Comment:
Note that to make this work most easily, you should move the call to this new successor function **below** the point in `remove_moderator()` where `ret = super(Subreddit, self).remove_moderator(user, **kwargs)` is called (so that the user being removed is no longer a mod, and won't be considered as a candidate, and you don't have to worry about skipping them or anything).